graylog2-web-interface/src/components/metrics/MetricContainer.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import String from 'string';
+
+import { Spinner } from 'components/common';
+
+import InjectStore from 'injection/InjectStore';
+import InjectActions from 'injection/InjectActions';
+
+import MetricsExtractor from 'logic/metrics/MetricsExtractor';
+
+var MetricContainer = React.createClass({
+  propTypes: {
+    name: React.PropTypes.string.isRequired,
+    zeroOnMissing: React.PropTypes.bool
+  },
+  mixins: [InjectActions('Metrics'), InjectStore('Metrics')],
+  componentWillMount() {
+    this.MetricsActions.addGlobal(this.props.name);
+  },
+  componentWillUnmount() {
+    this.MetricsActions.removeGlobal(this.props.name);
+  },
+  getDefaultProps() {
```

Comment:
`componentWillMount` and `componentWillUnmount` should go after `getDefaultProps` and `getInitialState`.